static/js/components/dashboard/ExamEnrollmentDialog.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+import React from "react"
+import PropTypes from "prop-types"
+import Dialog from "@material-ui/core/Dialog"
+import Button from "@material-ui/core/Button"
+
+import { singleBtnDialogActions } from "../inputs/util"
+import DialogTitle from "@material-ui/core/DialogTitle"
+import DialogActions from "@material-ui/core/DialogActions"
+import DialogContent from "@material-ui/core/DialogContent"
+import type { Course } from "../../flow/programTypes"
+
+export default class ExamEnrollmentDialog extends React.Component {
+  static contextTypes = {
+    router: PropTypes.object.isRequired
+  }
+
+  props: {
+    open: boolean,
+    course: ?Course,
+    setVisibility: (v: boolean) => void
+  }
+
+  handleCalculateClick = () => {
+    const { setVisibility } = this.props
+    setVisibility(false)
+  }
+
+  render() {
+    const { setVisibility, open, course } = this.props
+    if (!course) {
+      return null
+    }
+
+    const examRegisterButton = (
+      <Button
+        key="register-button"
+        onClick={() => {
+          course.exam_url
+            ? (window.location = course.exam_url)
+            : setVisibility(false)
+        }}
```

Comment:
If we could write it too in the helper function (function in the current component). If you feel that's okay.